src/homework10/hashmap/HashMap.java

```diff
@@ -0,0 +1,122 @@
+package homework10.hashmap;
+
+import java.util.NoSuchElementException;
+
+public class HashMap<K, V> {
+    private static final int INITIAL_SIZE = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static int size = 0;
+    private Entry<K, V>[] entries;
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+        private Entry<K, V> entryNext;
+
+        public Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+    }
+
+    public void put(K key, V value) {
+        if (isEmpty()) {
+            entries = new Entry[INITIAL_SIZE];
+        }
+
+        if (checkLoadEntries()) {
+            reSize();
+        }
+
+        int index = getIndexByHash(key);
+        Entry<K, V> newEntry = new Entry<>(key, value);
+
+        if (entries[index] == null) {
+            entries[index] = newEntry;
+        } else {
+            Entry<K, V> entry = entries[index];
+            if ((key == null && entry.getKey() == null) || entry.getKey().equals(key)) {
+                entry.setValue(value);
+                return;
+            }
+
+            entry = entry.entryNext;
```

Comment:
maybe better remove this line?
because in line 58 we skipped one element

Reply:
Ok, I agree with you.